### PR TITLE
feat(team): display team

### PR DESCRIPTION
**Pull request by elizachi**
The current note's team is now displayed in the settings

### CODE REVIEW

File: src/application/services/useTeam.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  /**
+   * Get all team members by note id
+   *
+   * @param id - Note id
+   */
+  const loadTeam = async (id: NoteId): Promise<void> => {
+    team.value = await teamService.getTeamByNoteId(id);
+  };
+
+  // onMounted(async() => {
```
Comment (slaveeks) [inline src/application/services/useTeam.ts:42]: Remove it

Comment (kloV148) [inline src/application/services/useTeam.ts:42]: Do we really need this commented block?

File: src/domain/entities/Team.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import type { NoteId } from './Note.js';
+
+/**
+ * Class representing a team entity
+ * Team is a relation between note and user, which shows what user can do with note
+ */
+export interface TeamMember {
```
Comment (slaveeks) [inline src/domain/entities/Team.ts:7]: Maybe add role here?

File: src/domain/noteSettings.repository.interface.ts [hunk 3]
```diff
@@ -30,4 +31,12 @@ export default interface NoteSettingsRepositoryInterface {
    * @returns updated note settings
    */
   regenerateInvitationHash(id: NoteId): Promise<NoteSettings>;
+
+  /**
+   * Returns all team members by note id
+   *
+   * @param  id - note  id
+   * @returns NoteSettings - NoteSettings instance
```
Comment (slaveeks) [inline src/domain/noteSettings.repository.interface.ts:39]: Wrong doc

File: src/domain/team.service.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import type NoteSettingsRepository from '@/domain/noteSettings.repository.interface';
+import type { Team } from './entities/Team';
+import type { NoteId } from './entities/Note';
+import NotFoundError from './entities/errors/NotFound';
+
+/**
+ * NoteList Service
```
Comment (slaveeks) [inline src/domain/team.service.ts:7]: Wrong doc :)

File: src/presentation/components/team/Member.vue [hunk 5]
```diff
@@ -0,0 +1,79 @@
+<template>
+  <div class="team">
```
Comment (slaveeks) [inline src/presentation/components/team/Member.vue:2]: As I remember we use css modules

File: src/presentation/components/team/Member.vue [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+}>();
+
+/**
+ * Get the title like userId
+ */
+const teamTitle = computed(() => {
+  const firstNoteBlock = props.teamMember.noteId;
+
+  if (!firstNoteBlock) {
+    return 'Loading...';
```
Comment (slaveeks) [inline src/presentation/components/team/Member.vue:32]: Use i18n

File: src/presentation/components/team/Member.vue [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  }
+});
+
+/**
+ * Get the subtitle depending on the type of block from Note
+ */
+const teamSubTitle = computed(() => {
+  const secondNoteBlock = props.teamMember.userId;
+
+  if (!secondNoteBlock) {
```
Comment (slaveeks) [inline src/presentation/components/team/Member.vue:44]: Use i18n here too

File: src/presentation/components/team/Member.vue [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+.note-header {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
+
+.note-title {
+  font-size: 24px;
```
Comment (slaveeks) [inline src/presentation/components/team/Member.vue:67]: We have vars for this

File: src/domain/entities/Team.ts (hunk 2 above)
Comment (elizachi) [reply src/domain/entities/Team.ts:7]: It is for next separate PR

File: src/domain/team.service.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+   * @param id - Note id
+   */
+  public async getTeamByNoteId(id: NoteId): Promise<Team> {
+    let result;
+
+    try {
+      result = await this.repository.getTeamByNoteId(id);
+    } catch (error) {
+      if (error instanceof NotFoundError) {
+        throw new Error(`404 Not Found: We couldn't find team of ${id} note.`);
```
Comment (slaveeks) [inline src/domain/team.service.ts:33]: I think, it can not be no team in existing note

File: src/presentation/components/team/Member.vue (hunk 5 above)
Comment (TatianaFomina) [reply src/presentation/components/team/Member.vue:2]: ➕

Comment (TatianaFomina) [reply src/presentation/components/team/Member.vue:2]: see how classes are used [here](https://github.com/codex-team/notes.web/blob/main/src/presentation/pages/Settings.vue#L8)

File: src/presentation/components/team/Member.vue (hunk 8 above)
Comment (TatianaFomina) [reply src/presentation/components/team/Member.vue:49]: use [typography](https://github.com/codex-team/notes.web/blob/main/codex-ui/src/styles/typography.pcss)

File: src/presentation/components/team/Member.vue [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+  return firstNoteBlock;
+});
+</script>
+
+<style>
+.note {
+  margin: 20px 0;
+  padding-bottom: 20px;
+  border-bottom: 1px solid #4b4b4b;
```
Comment (TatianaFomina) [inline src/presentation/components/team/Member.vue:38]: use color variables

File: src/application/services/useNoteSettings.ts [hunk 11]
```diff
@@ -17,15 +17,15 @@ interface UseNoteSettingsComposableState {
    *
    * @param id - note id
    */
-  load: (id: NoteId) => Promise<void>;
+  loadSettings: (id: NoteId) => Promise<void>;
```
Comment (TatianaFomina) [inline src/application/services/useNoteSettings.ts:20]: I guess you renamed functions to make it clear what they do in the place where they are called? No need to rename the functions themselves for that, you can do it when destructuring return value of `useNoteSettings`

```
const { load: loadSettings } = useNoteSettings();
```

File: src/presentation/components/team/Member.vue [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+ */
+const props = defineProps<{
+  /**
+   * Team member data
+   */
+  teamMember: TeamMember;
+}>();
+</script>
+
+<style scoped lang="postcss" module>
```
Comment (neSpecc) [inline src/presentation/components/team/Member.vue:23]: css modules are not used

File: src/presentation/components/team/Member.vue [hunk 13]
```diff
@@ -0,0 +1,33 @@
+<template>
+  <li class="member">
+    <div class="member-name">
+      {{ props.teamMember.user.name }}
```
Comment (neSpecc) [inline src/presentation/components/team/Member.vue:4]: props are available in template
```suggestion
      {{ teamMember.user.name }}
```

File: src/presentation/components/team/Team.vue [hunk 14]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <h1>{{ $t('team.title') }}</h1>
+  <ul
+    v-for="member in props.team"
```
Comment (neSpecc) [inline src/presentation/components/team/Team.vue:4]:
```suggestion
    v-for="member in team"
```

File: src/presentation/components/team/Member.vue [hunk 15]
```diff
@@ -0,0 +1,32 @@
+<template>
+  <li class="member">
+    <div class="member-name">
+      {{ teamMember.user.name }}
```
Comment (neSpecc) [inline src/presentation/components/team/Member.vue:4]: what if user does not have a name?

Comment (elizachi) [reply src/presentation/components/team/Member.vue:4]: As I know name it is field what cannot be empty

Comment (elizachi) [reply src/presentation/components/team/Member.vue:4]: But if this case may happend I can add someting like "user + user number" instead his name
Or create field nickname in the User entity

Comment (elizachi) [reply src/presentation/components/team/Member.vue:4]: What case do you think is the best?

Comment (slaveeks) [reply src/presentation/components/team/Member.vue:4]: Maybe use email? 
It is required for authentication

**Review by neSpecc — APPROVED** [review]

**Review by TatianaFomina — APPROVED** [review]

**Pull request merged**